File: sample/use_cocoapods/objectivec/GNAdGoogleBannerAdapterSample/GNAdGoogleBannerAdapterSample/ViewController.m
//
//  ViewController.m
//  GNAdGoogleBannerAdapterSample
//

@import GoogleMobileAds;

#import "ViewController.h"
#import "Util.h"

@interface ViewController() <GADBannerViewDelegate, UIPickerViewDelegate, UITextFieldDelegate>

@property (weak, nonatomic) IBOutlet UITextField *unitIdView;
@property (weak, nonatomic) IBOutlet UIPickerView *adSizeView;

@property(nonatomic, strong) GAMBannerView *bannerView;

@end

@implementation ViewController {
    NSMutableArray* adSizeArray;
    int adSizeIndex;
}

- (void)viewDidLoad {
    [super viewDidLoad];

    adSizeIndex = 0;
    _unitIdView.delegate = self;
    _adSizeView.delegate = self;
    [self setAdSizeArray];

    GADMobileAds.sharedInstance.requestConfiguration.testDeviceIdentifiers = @[[Util admobDeviceID]];
}

- (void)setAdSizeArray {
    adSizeArray = [NSMutableArray array];
    [adSizeArray addObject:@"Select AdSizeType"];
    [adSizeArray addObject:@"Banner"];              // 320x50
    [adSizeArray addObject:@"MediumRectangle"];     // 300x250
}

- (GADAdSize)getAdSizeType:(int)index {
    GADAdSize adSizeType = kGADAdSizeInvalid;
    switch (index) {
        case 1:
            adSizeType = kGADAdSizeBanner;
            break;
        case 2:
            adSizeType = kGADAdSizeMediumRectangle;
            break;
        default:
            adSizeType = kGADAdSizeInvalid;
            break;
    }
    return adSizeType;
}

- (IBAction)loadDownButton:(id)sender {
    if (_unitIdView.text.length == 0) {
        return;
    }
    if (adSizeIndex == 0) {
        return;
    }
    if (self.bannerView != nil) {
        [self.bannerView removeFromSuperview];
        self.bannerView.delegate = nil;
        self.bannerView = nil;
    }

    // Instantiate the banner view with your desired banner size.
    _bannerView = [[GAMBannerView alloc] initWithAdSize:[self getAdSizeType:adSizeIndex]];
    [self addBannerViewToView:self.bannerView];
    _bannerView.adUnitID = _unitIdView.text;
    _bannerView.delegate = self;
    _bannerView.rootViewController = self;
    GAMRequest * request = [GAMRequest request];
    [self.bannerView loadRequest:request];
}


#pragma mark - view positioning
-(void)addBannerViewToView:(UIView *_Nonnull)bannerView {
    self.bannerView.translatesAutoresizingMaskIntoConstraints = NO;
    [self.view addSubview:self.bannerView];

    [self.view addConstraint:[NSLayoutConstraint constraintWithItem:bannerView
                                                          attribute:NSLayoutAttributeCenterX
                                                          relatedBy:NSLayoutRelationEqual
                                                             toItem:self.view
                                                          attribute:NSLayoutAttributeCenterX
                                                         multiplier:1
                                                           constant:0]];
    [self.view addConstraint:[NSLayoutConstraint constraintWithItem:bannerView
                                                          attribute:NSLayoutAttributeBottom
                                                          relatedBy:NSLayoutRelationEqual
                                                             toItem:self.view
                                                          attribute:NSLayoutAttributeBottom
                                                         multiplier:1
                                                           constant:-20]];
}


#pragma mark GADBannerViewDelegate impl

// We've received an ad successfully.
- (void)bannerViewDidReceiveAd:(nonnull GADBannerView *)bannerView {
    NSLog(@"ViewController: bannerViewDidReceiveAd");
}

- (void)bannerView:(nonnull GADBannerView *)bannerView
        didFailToReceiveAdWithError:(nonnull NSError *)error {
    NSLog(@"ViewController: didFailToReceiveAdWithError: %@", [error localizedDescription]);
}


#pragma mark UIPickerViewDelegate.

- (NSInteger)numberOfComponentsInPickerView:(UIPickerView *)pickerView {
    return 1;
}

- (NSInteger)pickerView: (UIPickerView*)pView numberOfRowsInComponent:(NSInteger) component {
    NSInteger cnt = [adSizeArray count];
    return cnt;
}

- (NSString*)pickerView: (UIPickerView*) pView titleForRow:(NSInteger) row forComponent:(NSInteger)componet
{
    return [adSizeArray objectAtIndex:row];
}

- (void)pickerView:(UIPickerView *)pickerView didSelectRow:(NSInteger)row inComponent:(NSInteger)component
{
    adSizeIndex = (int)[pickerView selectedRowInComponent:0];
}


#pragma mark - UITextFieldDelegate
-(BOOL) textFieldShouldReturn:(UITextField *)textField {
    [textField resignFirstResponder];
    return YES;
}

@end

File: sample/use_cocoapods/swift/GNAdGoogleBannerAdapterSample/GNAdGoogleBannerAdapterSample/ViewController.swift
//
//  ViewController.swift
//  GNAdGoogleBannerAdapterSample
//

import UIKit
import GoogleMobileAds

class ViewController: UIViewController {

    @IBOutlet weak var unitIdView: UITextField!
    @IBOutlet weak var adSizeView: UIPickerView!
    
    private var adSizeList: [String] = []
    private var adSizeIndex = 0
    private var bannerView: GAMBannerView!

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        adSizeView.delegate = self
        adSizeView.dataSource = self
        unitIdView.delegate = self
        setAdSizeArray()

        GADMobileAds.sharedInstance().requestConfiguration.testDeviceIdentifiers = [Util.admobDeviceID()]
    }

    private func setAdSizeArray() {
        adSizeList.append("Select AdSizeType")
        adSizeList.append("Banner")             // 320x50
        adSizeList.append("MediumRectangle")    // 300x250
    }

    private func getAdSizeType(_ index: Int) -> GADAdSize {
        var adSizeType: GADAdSize = kGADAdSizeInvalid
        switch (index) {
            case 1:
                adSizeType = kGADAdSizeBanner
                break
            case 2:
                adSizeType = kGADAdSizeMediumRectangle
                break
            default:
                adSizeType = kGADAdSizeInvalid
                break
        }
        return adSizeType
    }

    @IBAction func loadDownButton(_ sender: Any) {
        if (unitIdView.text?.count == 0) {
            return
        }
        if (adSizeIndex == 0) {
            return
        }
        if (self.bannerView != nil) {
            self.bannerView.removeFromSuperview()
            self.bannerView.delegate = nil
            self.bannerView = nil
        }

        // Instantiate the banner view with your desired banner size.
        self.bannerView = GAMBannerView.init(adSize: getAdSizeType(adSizeIndex))
        self.addBannerViewToView(self.bannerView)
        self.bannerView.adUnitID = unitIdView.text
        self.bannerView.delegate = self
        self.bannerView.rootViewController = self
        let request: GAMRequest = GAMRequest.init()
        self.bannerView.load(request)
    }
    

    func addBannerViewToView(_ bannerView: UIView) {
        bannerView.translatesAutoresizingMaskIntoConstraints = false
        view.addSubview(bannerView)

        view.addConstraint(NSLayoutConstraint(item: bannerView,
                                              attribute: .centerX,
                                              relatedBy: .equal,
                                              toItem: view,
                                              attribute: .centerX,
                                              multiplier: 1,
                                              constant: 0))
        view.addConstraint(NSLayoutConstraint(item: bannerView,
                                              attribute: .bottom,
                                              relatedBy: .equal,
                                              toItem: view,
                                              attribute: .bottom,
                                              multiplier: 1,
                                              constant: -20))
    }

}
// MARK: - GADBannerViewDelegate
extension ViewController: GADBannerViewDelegate {
    /// Tells the delegate an ad request loaded an ad.
    func bannerViewDidReceiveAd(_ bannerView: GADBannerView) {
        print("ViewController: bannerViewDidReceiveAd")
    }

    /// Tells the delegate an ad request failed.
    func bannerView(_ bannerView: GADBannerView, didFailToReceiveAdWithError error: Error) {
        print("ViewController: didFailToReceiveAdWithError: \(error.localizedDescription)")
    }

}
// MARK: - UIPickerViewDataSource
extension ViewController: UIPickerViewDataSource {
    func numberOfComponents(in pickerView: UIPickerView) -> Int {
        return 1
    }

    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
        return adSizeList.count
    }

}
// MARK: - UIPickerViewDelegate
extension ViewController: UIPickerViewDelegate {
    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
        return adSizeList[row]
    }

    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
        adSizeIndex = pickerView.selectedRow(inComponent: 0)
    }

}
// MARK: - UITextFieldDelegate
extension ViewController: UITextFieldDelegate {
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
        // Hide the keyboard when press return key in a UITextField
        textField.resignFirstResponder()
        return true
    }

}
